core/src/main/kotlin/org/evomaster/core/search/service/monitor/StepOfSearchProcess.kt

```diff
@@ -0,0 +1,51 @@
+package org.evomaster.exps.monitor
+
+import org.evomaster.core.search.EvaluatedIndividual
+import org.evomaster.core.search.Individual
+import org.evomaster.core.search.service.Archive
+
+
+/**
+ * @author: manzhang
+ * @date: 2018/9/5
+ */
+open class StepOfSearchProcess<T> where T : Individual{
+
+    val populations = mutableMapOf<Int, MutableList<EvaluatedIndividual<T>>>()
+
+    val samplingCounter = mutableMapOf<Int, Int>()
```

Comment:
for maps, it is important to always have JavaDocs specifyin what are the keys, and what are the values